### PR TITLE
feat: select component

### PR DESCRIPTION
**Pull request by talamcol**
References [DM-389](https://autoricardo.atlassian.net/browse/DM-389)

## Motivation and context

For our brilliant selects on forms

## Before

No select

## After

Select

## How to test

1. compare Forms / Select with the [design](https://www.figma.com/file/CHU4fpMfXpiAW4qNFzi9y1/AS24-WebComponents?node-id=248%3A7307)

### CODE REVIEW

**Review by Marine-Berthier — APPROVED** [review]
It looks good to me, thank you 👍

**Review by Averethel — APPROVED** [review]
A few comments from my side, good to go once tackled:

File: src/themes/components/select.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import Input from './input';
+
+const selectStyles: StyleConfig = {
+  ...Input,
+  baseStyle: {
+    field: {
+      ...Input.baseStyle.field,
+    },
+    icon: {
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/themes/components/select.ts:11]: it is covered by the dropdown, so I guess that would be overkill.

File: src/components/icons/ArrowDownIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/icons/ArrowDownIcon.tsx:1]: We already have an ArrowLeft that is why I followed that pattern. I'll check if it's possible 👍

File: src/components/select/index.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      { value: 'o3', option: 'Option 3' },
+      { value: 'o4', option: 'Option 4' },
+      { value: 'o5', option: 'Option 5' },
+      { value: 'o6', option: 'Option 6' },
+    ],
+  }}
+  argTypes={{
+    value: {
+      defaultValue: '',
+      control: { type: 'text', placeholder: 'xxxxxx' },
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/select/index.stories.mdx:33]: sure. I'll adjust it for input as well then

File: src/components/select/index.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      { value: 'o2', option: 'Option 2' },
+      { value: 'o3', option: 'Option 3' },
+      { value: 'o4', option: 'Option 4' },
+      { value: 'o5', option: 'Option 5' },
+      { value: 'o6', option: 'Option 6' },
+    ],
+  }}
+  argTypes={{
+    value: {
+      defaultValue: '',
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/select/index.stories.mdx:32]: 👍 I'll change it for input as well

File: src/components/select/index.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  | 'isDisabled'
+  | 'onBlur'
+  | 'onFocus'
+  | 'onChange'
+  | 'autoFocus'
+> & {
+  size?: 'md' | 'lg';
+  value?: string | number;
+  name: string;
+  options: Option[];
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/select/index.tsx:27]: Hmmm how would that look if you use the component. Where do you pass the generic?

[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/select/index.tsx:27]: as discussed in slack :)

File: src/components/icons/ArrowDownIcon.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (talamcol) [reply src/components/icons/ArrowDownIcon.tsx:1]: I just realized, it's not an arrow and it's also not the same icon as the ArrowLeft :) it's a chevron.
However, rotating it with CSS is easily doable by doing this:
`<ArrowLeftIcon transform="rotate(90deg)" />`

**Pull request merged**